packages/loopring_v3/contracts/aux/access/LoopringIOExchangeOwner.sol

```diff
@@ -50,9 +53,48 @@ contract LoopringIOExchangeOwner is SelectorBasedAccessManager, ERC1271, Drainab
         address[]       receivers;
     }
 
-    constructor(address _exchange)
+    struct GasTokenConfig
+    {
+        uint maxToBurn;
+        uint expectedGasRefund;
+        uint calldataCost;
+    }
+
+    // See:
+    // - https://github.com/1inch-exchange/1inchProtocol/blob/a7781cf9aa1cc2aaa5ccab0d54ecbae1327ca08f/contracts/OneSplitAudit.sol#L343
+    // - https://github.com/curvefi/curve-ren-adapter/blob/8c1fbc3fec41ebd79b06984d72ff6ace3198e62d/truffle/contracts/CurveExchangeAdapter.sol#L104
+    modifier discountCHI(GasTokenConfig memory config)
+    {
+        uint gasStart = gasleft();
+        _;
+        if (config.maxToBurn == 0) return;
+        uint gasSpent = 21000 + gasStart - gasleft() + 14154;
+        gasSpent += (config.calldataCost == 0) ? 16 * msg.data.length : config.calldataCost;
+        uint gasRefundOffset = (config.expectedGasRefund * 2 > gasSpent) ? gasSpent : config.expectedGasRefund * 2;
```

Comment:
If `config.expectedGasRefund`== 0, then `fullAmountToBurn` will be `gasSpent/41947`. should it be `gasSpent /(2*41947)` instead as we can only refund up to 50% of `gasSpent`

Reply:
The 41947 already multiplies the gas refunded by 2 for the 50% limit. Every gas token theoretically refunds 24,000 gas (refund for a self destruct), but there are some additional overheads as well (a CALL most notably) so it's more around 21,000 gas. `41947` is the value used by 1inch and I kept it. CALL cost will increase soon in the HF so the constant will likely need updating then for best estmate.